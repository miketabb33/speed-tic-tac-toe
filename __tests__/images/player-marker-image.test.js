import renderer from 'react-test-renderer'
import PlayerMarkerImage from '../../images/player-marker-image'

var subject

const someLocation = "some-ui-location"
const someImageSize = 100

beforeEach(()=> {
  subject = new PlayerMarkerImage
})

test('x image rendering', () => {
  const rendering = renderer.create(
    subject.get('X', someImageSize, someLocation)
  )
  
  const result = rendering.toJSON()
	expect(result).toMatchSnapshot()
})

test('o image rendering', () => {
  const rendering = renderer.create(
    subject.get('O', someImageSize, someLocation)
  )
  
  const result = rendering.toJSON()
	expect(result).toMatchSnapshot()
})

test('red x image rendering', () => {
  const rendering = renderer.create(
    subject.get('redX', someImageSize, someLocation)
  )
  
  const result = rendering.toJSON()
	expect(result).toMatchSnapshot()
})

test('red o image rendering', () => {
  const rendering = renderer.create(
    subject.get('redO', someImageSize, someLocation)
  )
  
  const result = rendering.toJSON()
	expect(result).toMatchSnapshot()
})

test('yellow x image rendering', () => {
  const rendering = renderer.create(
    subject.get('yellowX', someImageSize, someLocation)
  )
  
  const result = rendering.toJSON()
	expect(result).toMatchSnapshot()
})

test('yellow o image rendering', () => {
  const rendering = renderer.create(
    subject.get('yellowO', someImageSize, someLocation)
  )
  
  const result = rendering.toJSON()
	expect(result).toMatchSnapshot()
})