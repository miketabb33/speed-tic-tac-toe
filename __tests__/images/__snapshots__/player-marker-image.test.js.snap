// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clear image rendering 1`] = `
<div
  style={
    Object {
      "boxSizing": "border-box",
      "display": "inline-block",
      "margin": 0,
      "maxWidth": "100%",
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "boxSizing": "border-box",
        "display": "block",
        "maxWidth": "100%",
      }
    }
  >
    <img
      alt=""
      aria-hidden={true}
      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAwIiBoZWlnaHQ9IjEwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
      style={
        Object {
          "border": "none",
          "display": "block",
          "margin": 0,
          "maxWidth": "100%",
          "padding": 0,
        }
      }
    />
  </div>
  <img
    alt="clear-marker-some-ui-location"
    data-nimg="intrinsic"
    decoding="async"
    draggable="false"
    id="clear-marker-some-ui-location"
    src="/_next/image?url=%2Fplayer-marker%2Fclear.png&w=256&q=75"
    srcSet="/_next/image?url=%2Fplayer-marker%2Fclear.png&w=128&q=75 1x, /_next/image?url=%2Fplayer-marker%2Fclear.png&w=256&q=75 2x"
    style={
      Object {
        "border": "none",
        "bottom": 0,
        "boxSizing": "border-box",
        "display": "block",
        "height": 0,
        "left": 0,
        "margin": "auto",
        "maxHeight": "100%",
        "maxWidth": "100%",
        "minHeight": "100%",
        "minWidth": "100%",
        "objectFit": undefined,
        "objectPosition": undefined,
        "padding": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 0,
      }
    }
  />
  <noscript>
    <img
      alt="clear-marker-some-ui-location"
      data-nimg="intrinsic"
      decoding="async"
      draggable="false"
      id="clear-marker-some-ui-location"
      loading="lazy"
      src="/_next/image?url=%2Fplayer-marker%2Fclear.png&w=256&q=75"
      srcSet="/_next/image?url=%2Fplayer-marker%2Fclear.png&w=128&q=75 1x, /_next/image?url=%2Fplayer-marker%2Fclear.png&w=256&q=75 2x"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
  </noscript>
</div>
`;

exports[`o image rendering 1`] = `
<div
  style={
    Object {
      "boxSizing": "border-box",
      "display": "inline-block",
      "margin": 0,
      "maxWidth": "100%",
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "boxSizing": "border-box",
        "display": "block",
        "maxWidth": "100%",
      }
    }
  >
    <img
      alt=""
      aria-hidden={true}
      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAwIiBoZWlnaHQ9IjEwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
      style={
        Object {
          "border": "none",
          "display": "block",
          "margin": 0,
          "maxWidth": "100%",
          "padding": 0,
        }
      }
    />
  </div>
  <img
    alt="O-marker-some-ui-location"
    data-nimg="intrinsic"
    decoding="async"
    draggable="false"
    id="O-marker-some-ui-location"
    src="/_next/image?url=%2Fplayer-marker%2FO.png&w=256&q=75"
    srcSet="/_next/image?url=%2Fplayer-marker%2FO.png&w=128&q=75 1x, /_next/image?url=%2Fplayer-marker%2FO.png&w=256&q=75 2x"
    style={
      Object {
        "border": "none",
        "bottom": 0,
        "boxSizing": "border-box",
        "display": "block",
        "height": 0,
        "left": 0,
        "margin": "auto",
        "maxHeight": "100%",
        "maxWidth": "100%",
        "minHeight": "100%",
        "minWidth": "100%",
        "objectFit": undefined,
        "objectPosition": undefined,
        "padding": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 0,
      }
    }
  />
  <noscript>
    <img
      alt="O-marker-some-ui-location"
      data-nimg="intrinsic"
      decoding="async"
      draggable="false"
      id="O-marker-some-ui-location"
      loading="lazy"
      src="/_next/image?url=%2Fplayer-marker%2FO.png&w=256&q=75"
      srcSet="/_next/image?url=%2Fplayer-marker%2FO.png&w=128&q=75 1x, /_next/image?url=%2Fplayer-marker%2FO.png&w=256&q=75 2x"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
  </noscript>
</div>
`;

exports[`red o image rendering 1`] = `
<div
  style={
    Object {
      "boxSizing": "border-box",
      "display": "inline-block",
      "margin": 0,
      "maxWidth": "100%",
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "boxSizing": "border-box",
        "display": "block",
        "maxWidth": "100%",
      }
    }
  >
    <img
      alt=""
      aria-hidden={true}
      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAwIiBoZWlnaHQ9IjEwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
      style={
        Object {
          "border": "none",
          "display": "block",
          "margin": 0,
          "maxWidth": "100%",
          "padding": 0,
        }
      }
    />
  </div>
  <img
    alt="redO-marker-some-ui-location"
    data-nimg="intrinsic"
    decoding="async"
    draggable="false"
    id="redO-marker-some-ui-location"
    src="/_next/image?url=%2Fplayer-marker%2FredO.png&w=256&q=75"
    srcSet="/_next/image?url=%2Fplayer-marker%2FredO.png&w=128&q=75 1x, /_next/image?url=%2Fplayer-marker%2FredO.png&w=256&q=75 2x"
    style={
      Object {
        "border": "none",
        "bottom": 0,
        "boxSizing": "border-box",
        "display": "block",
        "height": 0,
        "left": 0,
        "margin": "auto",
        "maxHeight": "100%",
        "maxWidth": "100%",
        "minHeight": "100%",
        "minWidth": "100%",
        "objectFit": undefined,
        "objectPosition": undefined,
        "padding": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 0,
      }
    }
  />
  <noscript>
    <img
      alt="redO-marker-some-ui-location"
      data-nimg="intrinsic"
      decoding="async"
      draggable="false"
      id="redO-marker-some-ui-location"
      loading="lazy"
      src="/_next/image?url=%2Fplayer-marker%2FredO.png&w=256&q=75"
      srcSet="/_next/image?url=%2Fplayer-marker%2FredO.png&w=128&q=75 1x, /_next/image?url=%2Fplayer-marker%2FredO.png&w=256&q=75 2x"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
  </noscript>
</div>
`;

exports[`red x image rendering 1`] = `
<div
  style={
    Object {
      "boxSizing": "border-box",
      "display": "inline-block",
      "margin": 0,
      "maxWidth": "100%",
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "boxSizing": "border-box",
        "display": "block",
        "maxWidth": "100%",
      }
    }
  >
    <img
      alt=""
      aria-hidden={true}
      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAwIiBoZWlnaHQ9IjEwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
      style={
        Object {
          "border": "none",
          "display": "block",
          "margin": 0,
          "maxWidth": "100%",
          "padding": 0,
        }
      }
    />
  </div>
  <img
    alt="redX-marker-some-ui-location"
    data-nimg="intrinsic"
    decoding="async"
    draggable="false"
    id="redX-marker-some-ui-location"
    src="/_next/image?url=%2Fplayer-marker%2FredX.png&w=256&q=75"
    srcSet="/_next/image?url=%2Fplayer-marker%2FredX.png&w=128&q=75 1x, /_next/image?url=%2Fplayer-marker%2FredX.png&w=256&q=75 2x"
    style={
      Object {
        "border": "none",
        "bottom": 0,
        "boxSizing": "border-box",
        "display": "block",
        "height": 0,
        "left": 0,
        "margin": "auto",
        "maxHeight": "100%",
        "maxWidth": "100%",
        "minHeight": "100%",
        "minWidth": "100%",
        "objectFit": undefined,
        "objectPosition": undefined,
        "padding": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 0,
      }
    }
  />
  <noscript>
    <img
      alt="redX-marker-some-ui-location"
      data-nimg="intrinsic"
      decoding="async"
      draggable="false"
      id="redX-marker-some-ui-location"
      loading="lazy"
      src="/_next/image?url=%2Fplayer-marker%2FredX.png&w=256&q=75"
      srcSet="/_next/image?url=%2Fplayer-marker%2FredX.png&w=128&q=75 1x, /_next/image?url=%2Fplayer-marker%2FredX.png&w=256&q=75 2x"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
  </noscript>
</div>
`;

exports[`x image rendering 1`] = `
<div
  style={
    Object {
      "boxSizing": "border-box",
      "display": "inline-block",
      "margin": 0,
      "maxWidth": "100%",
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "boxSizing": "border-box",
        "display": "block",
        "maxWidth": "100%",
      }
    }
  >
    <img
      alt=""
      aria-hidden={true}
      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAwIiBoZWlnaHQ9IjEwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
      style={
        Object {
          "border": "none",
          "display": "block",
          "margin": 0,
          "maxWidth": "100%",
          "padding": 0,
        }
      }
    />
  </div>
  <img
    alt="X-marker-some-ui-location"
    data-nimg="intrinsic"
    decoding="async"
    draggable="false"
    id="X-marker-some-ui-location"
    src="/_next/image?url=%2Fplayer-marker%2FX.png&w=256&q=75"
    srcSet="/_next/image?url=%2Fplayer-marker%2FX.png&w=128&q=75 1x, /_next/image?url=%2Fplayer-marker%2FX.png&w=256&q=75 2x"
    style={
      Object {
        "border": "none",
        "bottom": 0,
        "boxSizing": "border-box",
        "display": "block",
        "height": 0,
        "left": 0,
        "margin": "auto",
        "maxHeight": "100%",
        "maxWidth": "100%",
        "minHeight": "100%",
        "minWidth": "100%",
        "objectFit": undefined,
        "objectPosition": undefined,
        "padding": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 0,
      }
    }
  />
  <noscript>
    <img
      alt="X-marker-some-ui-location"
      data-nimg="intrinsic"
      decoding="async"
      draggable="false"
      id="X-marker-some-ui-location"
      loading="lazy"
      src="/_next/image?url=%2Fplayer-marker%2FX.png&w=256&q=75"
      srcSet="/_next/image?url=%2Fplayer-marker%2FX.png&w=128&q=75 1x, /_next/image?url=%2Fplayer-marker%2FX.png&w=256&q=75 2x"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
  </noscript>
</div>
`;

exports[`yellow o image rendering 1`] = `
<div
  style={
    Object {
      "boxSizing": "border-box",
      "display": "inline-block",
      "margin": 0,
      "maxWidth": "100%",
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "boxSizing": "border-box",
        "display": "block",
        "maxWidth": "100%",
      }
    }
  >
    <img
      alt=""
      aria-hidden={true}
      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAwIiBoZWlnaHQ9IjEwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
      style={
        Object {
          "border": "none",
          "display": "block",
          "margin": 0,
          "maxWidth": "100%",
          "padding": 0,
        }
      }
    />
  </div>
  <img
    alt="yellowO-marker-some-ui-location"
    data-nimg="intrinsic"
    decoding="async"
    draggable="false"
    id="yellowO-marker-some-ui-location"
    src="/_next/image?url=%2Fplayer-marker%2FyellowO.png&w=256&q=75"
    srcSet="/_next/image?url=%2Fplayer-marker%2FyellowO.png&w=128&q=75 1x, /_next/image?url=%2Fplayer-marker%2FyellowO.png&w=256&q=75 2x"
    style={
      Object {
        "border": "none",
        "bottom": 0,
        "boxSizing": "border-box",
        "display": "block",
        "height": 0,
        "left": 0,
        "margin": "auto",
        "maxHeight": "100%",
        "maxWidth": "100%",
        "minHeight": "100%",
        "minWidth": "100%",
        "objectFit": undefined,
        "objectPosition": undefined,
        "padding": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 0,
      }
    }
  />
  <noscript>
    <img
      alt="yellowO-marker-some-ui-location"
      data-nimg="intrinsic"
      decoding="async"
      draggable="false"
      id="yellowO-marker-some-ui-location"
      loading="lazy"
      src="/_next/image?url=%2Fplayer-marker%2FyellowO.png&w=256&q=75"
      srcSet="/_next/image?url=%2Fplayer-marker%2FyellowO.png&w=128&q=75 1x, /_next/image?url=%2Fplayer-marker%2FyellowO.png&w=256&q=75 2x"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
  </noscript>
</div>
`;

exports[`yellow x image rendering 1`] = `
<div
  style={
    Object {
      "boxSizing": "border-box",
      "display": "inline-block",
      "margin": 0,
      "maxWidth": "100%",
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "boxSizing": "border-box",
        "display": "block",
        "maxWidth": "100%",
      }
    }
  >
    <img
      alt=""
      aria-hidden={true}
      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAwIiBoZWlnaHQ9IjEwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
      style={
        Object {
          "border": "none",
          "display": "block",
          "margin": 0,
          "maxWidth": "100%",
          "padding": 0,
        }
      }
    />
  </div>
  <img
    alt="yellowX-marker-some-ui-location"
    data-nimg="intrinsic"
    decoding="async"
    draggable="false"
    id="yellowX-marker-some-ui-location"
    src="/_next/image?url=%2Fplayer-marker%2FyellowX.png&w=256&q=75"
    srcSet="/_next/image?url=%2Fplayer-marker%2FyellowX.png&w=128&q=75 1x, /_next/image?url=%2Fplayer-marker%2FyellowX.png&w=256&q=75 2x"
    style={
      Object {
        "border": "none",
        "bottom": 0,
        "boxSizing": "border-box",
        "display": "block",
        "height": 0,
        "left": 0,
        "margin": "auto",
        "maxHeight": "100%",
        "maxWidth": "100%",
        "minHeight": "100%",
        "minWidth": "100%",
        "objectFit": undefined,
        "objectPosition": undefined,
        "padding": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 0,
      }
    }
  />
  <noscript>
    <img
      alt="yellowX-marker-some-ui-location"
      data-nimg="intrinsic"
      decoding="async"
      draggable="false"
      id="yellowX-marker-some-ui-location"
      loading="lazy"
      src="/_next/image?url=%2Fplayer-marker%2FyellowX.png&w=256&q=75"
      srcSet="/_next/image?url=%2Fplayer-marker%2FyellowX.png&w=128&q=75 1x, /_next/image?url=%2Fplayer-marker%2FyellowX.png&w=256&q=75 2x"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
  </noscript>
</div>
`;
